src/test/java/hw2/base/BaseTest.java

```diff
@@ -0,0 +1,155 @@
+        assertTrue(element.isDisplayed());
+        assertEquals(element.getText(), username.getUsername());
+    }
+
+    //ex1 steps
+    public void assertItemsOnTheHeaderSection(WebElement assertionData, String expectedHeaders) {
+        assertTrue(assertionData.isDisplayed());
+        assertEquals(assertionData.getText(), expectedHeaders);
+    }
+
+    public void checkImagesExistence(List<WebElement> images) {
+        Assert.assertEquals(images.size(), 4);
+        for (WebElement img : images) {
+            Assert.assertTrue(img.isDisplayed());
+        }
+    }
+
+    public void checkTextBelowIcons(List<WebElement> texts) {
+        assertEquals(texts.size(), 4);
+        for (WebElement text : texts) {
+            assertTrue(text.isDisplayed());
+        }
+    }
+
+    public void assertFrameButton() {
+        WebElement frame = driver.findElement(By.id("frame"));
+        boolean existedFrame = frame.isEnabled();
+        Assert.assertTrue(existedFrame);
+    }
+
+    public void switchToTheFrameAndCheck() {
+        WebElement frame = driver.findElement(By.id("frame"));
+        driver.switchTo().frame(frame);
+        WebElement iFrameButton = driver.findElement(By.id("button-frame"));
+        assertTrue(iFrameButton.isDisplayed());
+    }
+
+    public void switchToOriginalWindowBack() {
+        driver.switchTo().defaultContent();
+    }
+
+    public void assertItemsOnTheLeftSection(WebElement assertionData, String expectedHeaders) {
+        assertTrue(assertionData.isDisplayed());
+        assertEquals(assertionData.getText(), expectedHeaders);
+    }
+
+    //ex2 steps
+    public void openDifferentElementsPage() {
+        WebElement element = driver.findElement((By.xpath("//a/span[contains(text(),'Service')]")));
+        element.click();
+        driver.findElement((By.xpath("//a[contains(text(), 'Different elements')]")));
+        element.click();
+        driver.get("https://jdi-testing.github.io/jdi-light/different-elements.html");
+    }
+
+    public void selectWaterCheckbox(String input) {
+        driver.findElement(By.xpath(String.format("//*[@class = 'checkbox-row']/label[contains(., '%s')]", input))).click();
+    }
+
+    public void selectWindCheckbox(String input) {
+        driver.findElement(By.xpath(String.format("//*[@class = 'checkbox-row']/label[contains(., '%s')]", input))).click();
+    }
+
+    public void selectRadio(String input) {
+        driver.findElement(By.xpath(String.format("//*[@class = 'checkbox-row']/label[contains(., '%s')]", input))).click();
+    }
+
+    public void selectDropdown(String input) {
+        driver.findElement((By.xpath("//select[@class = 'uui-form-element']"))).click();
+        driver.findElement((By.xpath(String.format("//*[@class = 'uui-form-element']/option[contains(., '%s')]", input)))).click();
+    }
+
+    public void assertCheckboxes(String checkbox, boolean status) {
```

Comment:
in this method you should accept that checkboxes with appropriate names are checked. What you assert?

assertTrue(status) ????

Again:
1. Find from all checkboxes **by checkbox label** only that checkboxes you need to assert
2. Get their state
3. Assert that state is checked